### PR TITLE
Convert Persistence docs into topics

### PR DESCRIPTION
**Pull request by JakeSCahill**
Moved all configuration information to a single page - this is now a single, self-contained topic. Everything that a developer needs to know to configure Persistence is in this file. By doing this, we remove linear dependencies in the previous pages such as handling-member-failures.adoc.

If this PR is approved, we can move one step further and define a template for these configuration topics, for example, we could use the following structure:

- Intro
- Glossary
- Quickstart
- Production Checklist
- Sections with advice for configuring specific behavior and how that behavior affects other parts of Hazelcast
- Configuration options (the reference docs)
- Full example

The other pages are now their own topics. For example, someone who has configured Persistence knows that they can trigger a force-start or a partial-start to recover from cluster restart failures. We assume this knowledge and link to the config doc for those that do not have it.

### CODE REVIEW

File: docs/modules/storage/pages/backing-up-persistence.adoc [hunk 1]
```diff
@@ -1,9 +1,13 @@
-= Backing Up Persisted Data
-:description: You can take snapshots of your persistence store (persisted files) to be able to copy the data onto other clusters without having to shut down your cluster. This process is called a hot backup.
+= Triggering a Backup of the Persistence Store
```
Comment (JakeSCahill) [inline docs/modules/storage/pages/backing-up-persistence.adoc:1]:
```suggestion
= Backing Up the Persistence Store
```

File: docs/modules/management/pages/cluster-utilities.adoc [hunk 2]
```diff
@@ -75,8 +75,8 @@ for all the other usages.
 You can use the script `hz-cluster-admin`, which comes with the Hazelcast package, to
 get/change the state of your cluster, to shutdown your cluster and
 to force your cluster to clean its persisted data and make a fresh start.
-The latter is the Force Start operation of Persistence.
-See the xref:storage:persistence.adoc#force-start[Force Start section].
+The latter is the Force-start operation of Persistence.
```
Comment (JakeSCahill) [inline docs/modules/management/pages/cluster-utilities.adoc:78]:
```suggestion
The latter is the force-start operation of Persistence.
```

File: docs/modules/maintain-cluster/pages/rest-api.adoc [hunk 3]
```diff
@@ -486,18 +486,18 @@ curl http://127.0.0.1:${PORT}/hazelcast/rest/management/cluster/version
 
 |[[partial-start-force-start]]**Enterprise commands**
 
-* _Partial starting the cluster:_
+* _Triggering a partial-start on the cluster:_
 +
-To partial start the cluster when persistence is enabled, use the following command:
+To trigger a partial-start when pPersistence is enabled, use the following command:
```
Comment (JakeSCahill) [inline docs/modules/maintain-cluster/pages/rest-api.adoc:491]:
```suggestion
To trigger a partial-start when Persistence is enabled, use the following command:
```

File: docs/modules/maintain-cluster/pages/rest-api.adoc [hunk 4]
```diff
@@ -486,18 +486,18 @@ curl http://127.0.0.1:${PORT}/hazelcast/rest/management/cluster/version
...
 [source,shell]
 ----
 curl --data "${CLUSTERNAME}&${PASSWORD}" http://127.0.0.1:${PORT}/hazelcast/rest/management/cluster/partialStart/
 ----
 +
-* _Force starting the cluster:_
+* _Triggering a force-start on the cluster:_
 +
-To force start the cluster when persistence is enabled, use the following command:
+To trigger a force-start when pPersistence is enabled, use the following command:
```
Comment (JakeSCahill) [inline docs/modules/maintain-cluster/pages/rest-api.adoc:500]:
```suggestion
To trigger a force-start when Persistence is enabled, use the following command:
```

File: docs/modules/fault-tolerance/partials/nav.adoc [hunk 5]
```diff
@@ -1,13 +1,12 @@
...
-*** xref:storage:recover-single-member.adoc[Handling Member Failures]
-*** xref:storage:encryption-at-rest.adoc[Encrypting Data at Rest]
+*** xref:storage:configuring-persistence.adoc[Configuration]
+*** xref:storage:backing-up-persistence.adoc[Backups]
+*** xref:storage:triggering-force-start.adoc[Force-start]
+*** xref:storage:triggering-partial-start.adoc[Partial-start]
 *** xref:storage:copying-persistence.adoc[Copying Data]
-*** xref:storage:backing-up-persistence.adoc[Backing Up Data]
 *** xref:storage:persistence-on-intel.adoc[Intel Optane DC]
 *** xref:storage:performance.adoc[Performance Benchmark]
```
Comment (Holmistr) [inline docs/modules/fault-tolerance/partials/nav.adoc:11]: Not a right place for the comment but a note - the quality of the images in "persistence performance" chapter is quite poor.

**Review by Holmistr — COMMENTED** [review]
I have some minor comments but overall, I like the improvement and the shit towards the topic based content.

I would double-check the links. I don't know if the issues I found are actual bugs or it's some "preliminary" state.

**Review by Serdaro — APPROVED** [review]

File: docs/modules/storage/pages/configuring-persistence.adoc [hunk 6]
```diff
@@ -45,16 +118,150 @@ PersistenceConfig PersistenceConfig = new PersistenceConfig()
...
 config.setPersistenceConfig(PersistenceConfig);
+
+MapConfig mapConfig = config.getMapConfig("test-map");
+mapConfig.getDataPersistenceConfig().setEnabled(true);
+config.addMapConfig(mapConfig);
 ----
 --
 ====
 
+== Planning
```
[the remark this answers is not in the document]
Comment (JakeSCahill) [reply docs/modules/storage/pages/configuring-persistence.adoc:129]: hmm yeah, lets go with production checklist 😄

**Pull request merged**